### ISSUE TITLE
Design of Transaction List View

### ISSUE THREAD

**Issue by georgeef** [moneymanagerex/mmex-ios#27]
The Transaction list view contains very little information per transaction, due to limited space. The same is true for other lists, but transactions contain more info which is of interest to the user.

The Detail View contains more (or all) info, however it is inconvenient for the users to switch back and forth between the two views (this creates some discontinuity in the browsing through the List View).

## Proposal

Add a button or something clickable on a transaction item, which when pressed creates more space for the selected transaction. The items below it are pushed further down, and in the new space some more information appears for the selected transaction (e.g., splits, tags, number/link to attachments, etc.). In this way users can quickly preview more information, without leaving the List View.

**Comment by guanlisheng** [comment]
Inspired by DimeApp, so far, it has 
1. Txn day and hour 
2. Category as Icon
3.  Payee
4. amount in account currency 
5. with color to indicate income and expenses 

I think first step might be to 
6. add account
7. amount in base amount .
8. Then total number would make sense.

**Comment by georgeef** [comment]
proposal:
- interchange the position of category and time (time left, category below payee)
- add the parent category (max two levels are shown); useful for sub-categories with same name
- if txn currency is the base currency, do not show the same amount/currency again

also:
- scale (number of decimal digits) does not work well

**Comment by guanlisheng** [comment]
#5 would address the category model issue, this model is still early today. 

interchange should be fine and easy to change, just a design.

plan to add an Account picker in the top-right corner, as a global filter.  multi-account design introduces too much complexity so far

**Comment by guanlisheng** [comment]
in terms of category icon,  it is based on some pre-defined name in https://github.com/moneymanagerex/database

we can also consider them during database creation.

**Comment by georgeef** [comment]
ok, i see. these are the default categories when a new database is created. in my test database i have changed them since long time and then i forgot the default categories.

in any case we need a feature to customize categories and their icons, so we also need to store the association to icons (there is no field for this in the database).

we could also add a setting like "Show icons in categories".

**Comment by guanlisheng** [comment]
yup, that is a nice-to-have feature after we finished the major piece

**Comment by Jnolcox** [comment]
The transactions list is through the 'Manage' tab, right?

It seems like this list is the 'Checking' tab from the bottom nav;

<img width="478" alt="image" src="https://github.com/user-attachments/assets/08a42e60-4e71-4219-82e8-48736f9d27c0">

---

### Proposal
- Rename bottom nav to 'Transactions'
- Add slide actions for transaction quick actions
  - Reconcile, Follow Up, Duplicate 

![image](https://github.com/user-attachments/assets/2bb2d24c-2b6e-4b5e-879c-fe9834240682)
